crewai_tools/tools/linkup/linkup_search_tool.py

```diff
@@ -1,49 +1,62 @@
 
-from pydantic import PrivateAttr
+class LinkupSearchToolSchema(BaseModel):
+    """Input for LinkupBaseTool."""
+   
+    query: str = Field(
+        ..., description="The query to search for."
+    )
 
-class LinkupSearchTool:
+class LinkupSearchTool(BaseTool):
     name: str = "Linkup Search Tool"
-    description: str = "Performs an API call to Linkup to retrieve contextual information."
-    _client: LinkupClient = PrivateAttr() # type: ignore
+    description: str = "A tool to search and retrieve trends or insights using the Linkup API."
+    args_schema: Type[BaseModel] = LinkupSearchToolSchema
 
-    def __init__(self, api_key: str):
-        """
-        Initialize the tool with an API key.
-        """
+    def __init__(self, api_key: str, depth: str, output_type: str, structured_output_schema:str = None, **kwargs):
+        from linkup import LinkupClient
+
+        super().__init__(**kwargs)
         if not LINKUP_AVAILABLE:
             raise ImportError(
                 "The 'linkup' package is required to use the LinkupSearchTool. "
-                "Please install it with: uv add linkup"
+                "Please install it with: pip install linkup"
             )
         self._client = LinkupClient(api_key=api_key)
+        self._default_depth = depth
+        self._default_output_type = output_type
+        self._default_structured_schema = structured_output_schema
+
+    def _run(self, query: str ) :
 
-    def _run(self, query: str, depth: str = "standard", output_type: str = "searchResults") -> dict:
         """
         Executes a search using the Linkup API.
 
         :param query: The query to search for.
-        :param depth: Search depth (default is "standard").
-        :param output_type: Desired result type (default is "searchResults").
-        :return: A dictionary containing the results or an error message.
+        :return: The results or an error message.
         """
+
+        api_params = {
+            "query": query,
+            "depth": self._default_depth,
+            "output_type": self._default_output_type,
+        }
+
+        if self._default_output_type == "structured":
+            if not structured_output_schema:
+                raise ValueError("structured_output_schema must be provided when output_type is 'structured'.")
+            api_params["structured_output_schema"] = self._default_structured_schema
         try:
-            response = self._client.search(
-                query=query,
-                depth=depth,
-                output_type=output_type
-            )
-            results = [
-                {"name": result.name, "url": result.url, "content": result.content}
-                for result in response.results
-            ]
-            return {"success": True, "results": results}
+            return self._client.search(**api_params)
         except Exception as e:
-            return {"success": False, "error": str(e)}
+            return {"error": str(e)}
```

Comment:
Let's remove the try...catch and let the SDK throw a LinkupError if something goes wrong.